docs/api-reference/static-analysis.md

```diff
@@ -40,109 +45,219 @@ npm install --save-dev @dbos-inc/eslint-plugin
 
-Calls to functions such as `Math.random()` are [nondeterministic](../tutorials/workflow-tutorial#determinism), and may interfere with consistent workflow results or the debugger.
+1. Awaiting in a workflow on a non-`WorkflowContext` object (this implies I/O, which is often nondeterministic):
 
-Such operations should use functions provided by DBOS Transact, or at a minimum, be encapsulated in a [Communicator](../tutorials/communicator-tutorial).
+Not allowed:
 
-This rule is enabled by default in all `@dbos-inc/eslint-plugin` configurations.
+```ts
+@Workflow()
+static async myWorkflow(ctxt: WorkflowContext) {
+  const result = await fetch("https://www.google.com");
+}
+```
 
-#### `@dbos-inc/detect-new-date`
+Allowed:
 
-Calls to functions such as `new Date()` are [nondeterministic](../tutorials/workflow-tutorial#determinism), and may interfere with transactional data, consistent workflow execution, or the debugger.
+```ts
+@Communicator()
+static async myCommunicator(ctxt: CommunicatorContext) {
+  const result = await fetch("https://www.google.com");
+}
+```
 
-Such operations should use functions provided by DBOS Transact, or at a minimum, be encapsulated in a [Communicator](../tutorials/communicator-tutorial).
+```ts
+@Workflow()
+static async myWorkflow(ctxt: WorkflowContext) {
+  const user = await ctxt.client<User>('users').select("password").where({ username }).first();
+}
+```
 
-This rule is enabled by default in all `@dbos-inc/eslint-plugin` configurations.
+```ts
+@Workflow()
+static async myWorkflow(ctxt: WorkflowContext) {
+  /* Sometimes, you might await on a non-`WorkflowContext` object,
+  but the function you're calling is a helper function that uses the
+  underlying context. So if you pass in a `WorkflowContext` as a parameter,
+  the warning will be supressed. */
+  const user = await getUser(ctxt, username);
+}
+```
 
-#### `@dbos-inc/detect-native-code`
+2. Global modification in a workflow (this often leads to nondeterministic behavior):
 
-Code written in languages such as C/C++ that do not have automatic initialization, memory protection, etc., are at risk of a number of classes of bugs and security vulnerabilities that are not of concern in high-level languages like TypeScript.  C/C++ code must also be compiled for the target CPU architecture and Operating System.
+```ts
 
-Libraries that contain such native code present deployment challenges at best, and have a higher risk of security holes at worst.  Hence, DBOS suggests use of libraries that do not rely on native code.  For example, `bcryptjs` should be used instead of `bcrypt`.
+let globalUser = undefined;
 
-While DBOS cloud deployment prevents use of native code, use of the `@dbos-inc/detect-native-code` lint rule can catch usage of some libraries early in the coding process and suggest alternatives.
+@Workflow()
+static async myWorkflow(ctxt: WorkflowContext) {
+  globalUser = await ctxt.client<User>('users').select("password").where({ username }).first();
```

Comment:
This is a transaction?  I'd also use a simpler example, one that really is a workflow.

Reply:
Done